the-blue-alliance-ios/View Controllers/Teams/TeamSummaryTableViewController.swift

```diff
@@ -1,16 +1,320 @@
 import Foundation
 import UIKit
+import TBAKit
+import CoreData
+
+enum TeamSummaryRow: Int {
+    case rank
+    case awards
+    case pit // only during CMP, and if they exist
+    case record // don't show record for 2015, because no wins
+    case alliance
+    case status
+    case breakdown
+    case nextMatch
+    case lastMatch
+    case max
+}
 
 class TeamSummaryTableViewController: TBATableViewController {
+    
+    var team: Team!
+    var event: Event!
 
-    // Handle District and Event
+    private var eventStatus: EventStatus? {
+        didSet {
+            if let eventStatus = eventStatus {
+                updateSummaryInfo()
+
+                contextObserver.observeObject(object: eventStatus, state: .updated) { [weak self] (_, _) in
+                    self?.updateSummaryInfo()
+                }
+            } else {
+                contextObserver.observeInsertions { [weak self] (eventStatuses) in
+                    self?.eventStatus = eventStatuses.first
+                }
+            }
+        }
+    }
+    
+    // MARK: - Persistable
+    
+    override var persistentContainer: NSPersistentContainer! {
+        didSet {
+            eventStatus = EventStatus.findOrFetch(in: persistentContainer.viewContext, matching: observerPredicate)
+        }
+    }
+    
+    // MARK: - Observable
+    
+    typealias ManagedType = EventStatus
+    lazy var observerPredicate: NSPredicate = {
+        return NSPredicate(format: "event == %@ AND team == %@", event, team)
+    }()
+    lazy var contextObserver: CoreDataContextObserver<EventStatus> = {
+        return CoreDataContextObserver(context: persistentContainer.viewContext)
+    }()
+
+    var summaryRows: [TeamSummaryRow] = []
+    var summaryValues: [Any] = []
+    
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.register(UINib(nibName: String(describing: ReverseSubtitleTableViewCell.self),bundle: nil),
+                           forCellReuseIdentifier: ReverseSubtitleTableViewCell.reuseIdentifier)
+        tableView.register(UINib(nibName: String(describing: LoadingTableViewCell.self),bundle: nil),
+                           forCellReuseIdentifier: LoadingTableViewCell.reuseIdentifier)
+        tableView.register(UINib(nibName: String(describing: MatchTableViewCell.self),bundle: nil),
+                           forCellReuseIdentifier: MatchTableViewCell.reuseIdentifier)
+    }
+
+    // MARK: - Private
+    
+    private func fetchMatch(key: String, completion: @escaping (Match?) -> ()) {
```

Comment:
Use `TBABackgroundService.backgroundFetchTeam`

Reply:
fwiw we'll need to keep references to if we're currently downloading these... I guess we could set bools. Kinda sucks but